metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -205,16 +205,6 @@ def _build_metrics_output_node(
                     ),
                     metric_specs=[metric_spec],
                 )
-                if metric_spec.offset_window or metric_spec.offset_to_grain:
```

Comment:
Updated this logic to handle an offset in any metric, not just a derived metric. This is because this function will run for the input metric, not the derived metric. (Config restrictions will still prevent non-derived metrics from having an offset.)